src/native-common/Image.tsx

```diff
@@ -202,7 +202,16 @@ export class Image extends React.Component<Types.ImageProps, ImageState> impleme
 
         const source: RN.ImageSourcePropType = { uri: this.props.source };
         if (this.props.headers) {
-            source.headers = this.props.headers;
+            const cacheControlHeader = this._getMaxStaleHeader();
+            if (cacheControlHeader) {
+                // Filter out Cache-Control: max-stale. It has the opposite effect on iOS: instead of having
+                // the cache return stale data it disables the cache altogether. We emulate the header by
+                // retrying with cache: 'only-if-cached'.
+                source.headers = _.clone(this.props.headers);
```

Comment:
nit: `_buildSource` is called directly in the `render` method. For images which have headers and the header `Cache-Control: max-stale` `_.clone` method will be called on each re-render. For instance, I want to build an application like ImageList, which will have 50 images on a page with the header `Cache-Control: max-stale`, for this case, the method `clone` will be called 50 times, it will take some time.

Reply:
Gotcha, yes, this code doesn't have to run on each render. I have added headers to state. Thanks!